yb-voyager/cmd/importData.go

```diff
@@ -915,54 +896,112 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 	}
 }
 
-func executeSqlFile(file string) {
-	log.Infof("Execute SQL file %q on target %q", file, target.Host)
-	connectionURI := target.GetConnectionUri()
-	conn, err := pgx.Connect(context.Background(), connectionURI)
+func newTargetConn() *pgx.Conn {
+	conn, err := pgx.Connect(context.Background(), target.GetConnectionUri())
 	if err != nil {
 		utils.WaitChannel <- 1
 		<-utils.WaitChannel
 		utils.ErrExit("connect to target db: %s", err)
 	}
-	defer conn.Close(context.Background())
 
+	setTargetSchema(conn)
+	return conn
+}
+
+func setTargetSchema(conn *pgx.Conn) {
 	if sourceDBType != POSTGRESQL && target.Schema != YUGABYTEDB_DEFAULT_SCHEMA {
 		setSchemaQuery := fmt.Sprintf("SET SCHEMA '%s'", target.Schema)
 		_, err := conn.Exec(context.Background(), setSchemaQuery)
 		if err != nil {
 			utils.ErrExit("run query %q on target %q: %s", setSchemaQuery, target.Host, err)
 		}
 	}
+}
+
+func dropIdx(conn *pgx.Conn, idxName string) {
+	dropIdxQuery := fmt.Sprintf("DROP INDEX IF EXISTS %s", idxName)
+	log.Infof("Dropping index: %q", dropIdxQuery)
+	_, err := (*conn).Exec(context.Background(), dropIdxQuery)
```

Comment:
No need of dereferencing the `conn` pointer in the `(*conn).Exec`. Just `conn.Exec()` is sufficient.

Reply:
oh ok, that because its a struct

Reply:
done